lib/mix/lib/mix.ex

```diff
@@ -1013,6 +1056,21 @@ defmodule Mix do
     end
   end
 
+  @doc """
+  Returns the directory where the current `Mix.install/2` project
+  resides.
+  """
+  @spec install_project_dir() :: Path.t()
+  def install_project_dir() do
+    case Mix.State.get(:installed) do
+      {id, _dynamic_config} ->
+        install_dir(id)
+
+      nil ->
+        Mix.raise("trying to call Mix.install_project_dir/0, but Mix.install/2 was never called")
```

Comment:
No need to raise:

```suggestion
        nil
```

Then we only need to call this function to know if install ran or not.